armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -310,3 +368,179 @@ def _insertDepletableNuclideKeys(c, blueprint):
         dimName,
         yamlize.Attribute(name=dimName, type=ComponentDimension, default=None),
     )
+
+
+class _ParticleFuelLayer(yamlize.Object):
+    """Component-like specification for a single layer in particle fuel"""
+
+    name = yamlize.Attribute(key="name", type=str)
+    material = yamlize.Attribute(type=str)
+    innerDiam = yamlize.Attribute(key="id", type=float, default=0)
+    od = yamlize.Attribute(key="od", type=float)
+    Tinput = yamlize.Attribute(type=float)
+    Thot = yamlize.Attribute(type=float)
+    # Need this to pick up flags like depletable
+    flags = yamlize.Attribute(type=str, default=None)
```

Comment:
this is so tantalizingly close to a [sphere](https://github.com/terrapower/armi/blob/9eaa6dc95a1f3d564140b6d632dad482aed0e35b/armi/reactor/components/volumetricShapes.py#L23) that I really wonder if we could use it, probably as a subclass (`ParticleFuelSphere`) that has the extra code you have below. 

Advantages might include needing less code to set it up in general (since shared w/ component and it's blueprints), and the fact that you could either use blueprints or not more directly (e.g. for programmatically building models with fewer blueprints)